components/brave_rewards/browser/file_util.cc

```diff
@@ -0,0 +1,151 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/file_util.h"
+
+#include <memory>
+
+#include "base/logging.h"
+#include "base/strings/string_util.h"
+
+namespace brave_rewards {
+
+namespace {
+
+bool SeekLine(
+    base::File* file,
+    int64_t* offset) {
+  DCHECK(file);
+  DCHECK(offset);
+
+  char buffer[1];
+
+  do {
+    if (file->ReadAtCurrentPos(buffer, 1) == -1) {
```

Comment:
It's usually more efficient to read a fixed block size and then scan the buffer character by character vs. reading the file character by character.

Reply:
Fixed